QuizDifferently.js

```diff
@@ -1,67 +1,101 @@
+  // Return the array sorted by suit and rank
+  // Disclaimer: This is a tricky one to do with JavaScript
+  sortDeckOfCards(cards) {
+    return cards;
+  }
 
-    // Question 4
-    // Given a pile of socks, find how many socks are missing a pair.
-    // To represent the pile of socks, an array containing all the socks will be provided. Each sock will be represented by its color.
-    // Ex. ["Brown", "Blue", "Black", "Black", "Green", "Blue"]
-    // Return the number of socks without a match
-    missingMatches(socks){
-        return -1;
-    }
+  // Question 4
+  // Given a pile of socks, find how many socks are missing a pair.
+  // To represent the pile of socks, an array containing all the socks will be provided. Each sock will be represented by its color.
+  // Ex. ["Brown", "Blue", "Black", "Black", "Green", "Blue"]
+  // Return the number of socks without a match
+  missingMatches(socks) {
+    return -1;
+  }
 
-    //Question 5
-    // Given a roster of names, remove the duplicates from the roster
-    // To represent the roster, an array containing all the names will be provided.
-    // Ex. ["Daniel", "Kaleb", "Zan", "Mark", "Tariq", "Mark"]
-    // Return the an arrary without any duplicates
-    removeDuplicates(names){
-        return names;
+  //Question 5
+  // Given a roster of names, remove the duplicates from the roster
+  // To represent the roster, an array containing all the names will be provided.
+  // Ex. ["Daniel", "Kaleb", "Zan", "Mark", "Tariq", "Mark"]
+  // Return the an arrary without any duplicates
+  removeDuplicates(names) {
+    let rosterSet = new Set();
+    let noDupsRoster = [];
+
+    //insert roster of names to a Set object
+    //no duplicates will be inserted into set object
+    for (let i = 0; i < names.length; i++) {
+      rosterSet.add(names[i]);
     }
+    //copy each name from the set into a new no duplicate name roster, "noDupsRoster"
+    rosterSet.forEach((name) => {
+      noDupsRoster.push(name);
+    });
+
+    //return names from no duplicate name roster, "noDupsRoster"
+    return noDupsRoster;
+  }
 
-    // Question 6
-    // Start with a list of numbers from 1 to 100. I remove a random number from the list and shuffle the list. Find the number I removed.
-    // To represent the list of numbers, you will recieve an array of numbers
-    // Ex. [3,4,1,5] with the missing value, and the answer, being 2
-    // Return the missing value
-    missingNumber(numbers){
-        return -1
+  // Question 6
+  // Start with a list of numbers from 1 to 100. I remove a random number from the list and shuffle the list. Find the number I removed.
+  // To represent the list of numbers, you will recieve an array of numbers
+  // Ex. [3,4,1,5] with the missing value, and the answer, being 2
+  // Return the missing value
+  missingNumber(numbers) {
+    let sumList = 0;
+    let sumNumbers = 0;
+    let missingNum = 0;
+
+    //sum numbers from 1 to 100
+    for (let i = 1; i <= 100; i++) {
```

Comment:
There is a formula to calculate the sum from 1 to n. Take a look and try to implement it here. I for loop works but the formula works in constant time

Reply:
Thanks for your feedback Kaleb. I appreciate it. I will implement your suggestions.